src/phonex.rs

```diff
@@ -0,0 +1,344 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+use serde::{Deserialize, Serialize};
+
+use crate::helper::is_vowel;
+use crate::{Encoder, SoundexUtils};
+
+/// Phonex is a modification of the venerable Soundex algorithm. It accounts
+/// for a few more letter combinations to improve accuracy on some data sets.
+/// It was created by A.J. Lait and Brian Randell in 1996, described in their
+/// paper "An assessment of name matching algorithms" in the Technical Report
```

Comment:
I add the link to the paper in documentation, if anyone want to read it.
```suggestion
/// paper ["An assessment of name matching algorithms"](https://citeseerx.ist.psu.edu/viewdoc/download;jsessionid=E3997DC51F2046A95EE6459F2B997029?doi=10.1.1.453.4046&rep=rep1&type=pdf) in the Technical Report
```

Reply:
I'll directely commit suggestion and it's good !